File: python/app/common/result_code.py
from enum import Enum

class ResultCode(Enum):
    """响应码枚举"""
    SUCCESS = (200, "success")
    BAD_REQUEST = (400, "请求参数错误")
    UNAUTHORIZED = (401, "未授权，请先登录")
    FORBIDDEN = (403, "无权限访问")
    NOT_FOUND = (404, "资源不存在")
    INTERNAL_SERVER_ERROR = (500, "服务器内部错误")
    
    # 业务错误码
    USER_NOT_FOUND = (1001, "用户不存在")
    USER_ALREADY_EXISTS = (1002, "用户已存在")
    INVALID_CREDENTIALS = (1003, "邮箱或密码错误")
    TOKEN_INVALID = (1004, "Token无效或已过期")
    
    INSCRIPTION_NOT_FOUND = (2001, "碑文不存在")
    INSCRIPTION_ALREADY_EXISTS = (2002, "碑文已存在")
    
    KNOWLEDGE_NOT_FOUND = (3001, "知识库内容不存在")
    
    def __init__(self, code: int, message: str):
        self.code = code
        self.message = message

File: src/main/java/com/beishuo/common/ResultCode.java
package com.beishuo.common;

public enum ResultCode {
    SUCCESS(200, "success"),
    BAD_REQUEST(400, "请求参数错误"),
    UNAUTHORIZED(401, "未授权，请先登录"),
    FORBIDDEN(403, "无权限访问"),
    NOT_FOUND(404, "资源不存在"),
    METHOD_NOT_ALLOWED(405, "请求方法不允许"),
    INTERNAL_SERVER_ERROR(500, "服务器内部错误"),
    SERVICE_UNAVAILABLE(503, "服务暂不可用"),

    // 业务错误码
    USER_NOT_FOUND(1001, "用户不存在"),
    USER_ALREADY_EXISTS(1002, "用户已存在"),
    INVALID_CREDENTIALS(1003, "用户名或密码错误"),
    TOKEN_EXPIRED(1004, "Token已过期"),
    TOKEN_INVALID(1005, "Token无效"),

    INSCRIPTION_NOT_FOUND(2001, "碑文不存在"),
    INSCRIPTION_UPLOAD_FAILED(2002, "碑文图片上传失败"),
    INSCRIPTION_RECOGNITION_FAILED(2003, "碑文识别失败"),

    KNOWLEDGE_NOT_FOUND(3001, "知识库内容不存在"),

    FILE_UPLOAD_FAILED(4001, "文件上传失败"),
    FILE_TYPE_NOT_ALLOWED(4002, "文件类型不允许"),
    FILE_SIZE_EXCEEDED(4003, "文件大小超出限制");

    private final int code;
    private final String message;

    ResultCode(int code, String message) {
        this.code = code;
        this.message = message;
    }

    public int getCode() {
        return code;
    }

    public String getMessage() {
        return message;
    }
}

